app/controllers/batch_controller.rb

```diff
@@ -0,0 +1,37 @@
+class BatchController < ApplicationController
+  def create
+    uris = payload.fetch(:uris)
+    checked_within = (payload[:checked_within] || 24.hours).to_i
+    callback_uri = payload[:callback_uri]
+
+    return render json: { error: { message: "No URIs given." } }, status: 400 if uris.empty?
+
+    links = uris.map do |uri|
+      Link.find_or_create_by(uri: uri)
+    end
+
+    checks = links.map do |link|
+      check = link.find_completed_check(within: checked_within)
+      check ? check : Check.create!(link: link)
+    end
+
+    batch = Batch.create!(checks: checks)
+
+    if batch.completed?
+      WebhookJob.perform_now(batch, callback_uri) if callback_uri
```

Comment:
I imagine we probably want to still run this asynchronously since the IO could hold up our response and we don't use any of the response from this to complete our response